src/main/java/seedu/address/model/property/client/Client.java

```diff
@@ -57,12 +57,36 @@ public boolean isSameClient(Client otherClient) {
      * Converts toString of client back to Client object.
      */
     public static Client fromStringToClient(String toString) {
-        String[] clientInfo = toString.split(";");
-        String name = clientInfo[0].split("Client Name: ")[1];
-        String contact = clientInfo[1].split("Client Contact: ")[1];
-        String email = clientInfo[2].split("Client Email: ")[1];
-        String price = clientInfo[3].split("Client Asking Price: ")[1];
-        return new Client(new Name(name), new Contact(contact), new Email(email), new AskingPrice(price));
+        Name name = null;
+        Contact contact = null;
+        Email email = null;
+        AskingPrice price = null;
+
+        if (toString.contains("Client Name: ")) {
+            int start = toString.indexOf("Client Name: ") + 13;
```

Comment:
consider use of split function for more concise code?

Reply:
that would be one way of doing it, i just find this way easier

Reply:
okk I see that u updated the function to use length function 👍 I just found the '13' and other indexes to be random